src/axom/slic/interface/c_fortran/wrapfslic.f

```diff
@@ -20,12 +20,12 @@ module axom_slic
     ! splicer begin module_top
     ! splicer end module_top
 
-    !  Level
-    integer(C_INT), parameter :: level_error = 0
-    integer(C_INT), parameter :: level_warning = 1
-    integer(C_INT), parameter :: level_info = 2
-    integer(C_INT), parameter :: level_debug = 3
-    integer(C_INT), parameter :: level_num_levels = 4
+    !  enum axom::slic::message::Level
+    integer(C_INT), parameter :: message_error = 0
+    integer(C_INT), parameter :: message_warning = 1
+    integer(C_INT), parameter :: message_info = 2
+    integer(C_INT), parameter :: message_debug = 3
+    integer(C_INT), parameter :: message_num_levels = 4
```

Comment:
Is there a way to make these `slic_message_error` etc. That way, the C interface is mirrored more accurately?

Reply:
This does mirror C++ usage.  Here's an example from slic

    addStreamToMsgLevel(wefStream,  message::Error);

In Fortran `message::Error` becomes `message_error`.  If it was a class enum, then the class name would be included in the Fortran name similar to how the class name would be needed in C++.  Shroud is just "flattening" the namespace.

Reply:
I think in your C++ usage example, assumes `using namespace slic;`. Without this, you would need to have:
```
slic::addStreamToMsgLevel(wefStream, slic::message::Error)
```

Moreover, what prompted my comment was that in the auto-generate C file, we have this:
```
//  axom::slic::message::Level
enum SLIC_Level	enum SLIC_message_Level
{	{
  SLIC_Error,	  SLIC_message_Error,
  SLIC_Warning,	  SLIC_message_Warning,
  SLIC_Info,	  SLIC_message_Info,
  SLIC_Debug,	  SLIC_message_Debug,
  SLIC_Num_Levels	  SLIC_message_Num_Levels
};	};
```

So, I am wondering if the Fortran and C usage could be made more consistent, or if there is a better way to make C++/C/Fortran usage overall more consistent.

Reply:
The Fortran names are in a module so it's similar to `using namespace slic;`

    use slic
    call add_stream_to_msg_level(wefStream, message_error)

There is no way to access the names without using the module.  There have been suggestions to allow fully qualified names in Fortran similar to C++i.e. `slic%message_error`.  But if that happens it's years away.  In the case of name conflicts between modules, there are two current solutions.  `use slic, only : message_error` or `use slic, my_error => message_error`.

The C names are global, so they are always the fully flattened name with the prefix `SLIC_`.

Reply:
Thanks for the explanation. Yes the potential conflicts was the main thing that I was worried about. Sounds like we have workarounds, which is good.

 Using a fully qualified name would be nice and avoid these issues when it happens.